### PR TITLE
Spike/remove SupersetDB

### PR DESCRIPTION
**Pull request by dervoeti**
# Description

As discussed in https://github.com/stackabletech/superset-operator/issues/351:

 - Removed `.spec.clusterConfig.loadExamplesOnInit` (loading examples is not supported anymore by the operator, instead we want to document how it can be done)
 - Removed all the SupersetDB related code
 - Moved database init commands from a separate Job to the command of each Superset Pod (We can move it into an initContainer, needs discussion and probably a separate issue for that)
- Added liveness and readiness probes (I used these values: https://github.com/apache/superset/blob/35eb66a322f7938f840778633a4aea11c7f24dce/helm/superset/values.yaml#L300-L317)
 - Changed Pod Management Policy to `OrderedReady`, to make sure Pods start after another and the init commands don't run in parallel
 - It's still possible that during an upgrade from e.g. 1.5.3 to 2.1.0 some Pods temporarily are on 1.5.3 while others are already on 2.1.0. I don't think that's a problem though.
 - Wait for `SupersetCluster` instead of `SupersetDB` in DruidConnection

I tested various scenarios manually (multiple replicas, different upgrade paths, also the DruidConnection stuff), everything worked as expected. 

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [ ] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Comment by fhennig** [comment]
Is there also a ticket somewhere for adding instructions how to install the demo data again?

**Comment by dervoeti** [comment]
Will create one tomorrow, we'll discuss this briefly in the Arch meeting.

**Comment by dervoeti** [comment]
There it is: https://github.com/stackabletech/superset-operator/issues/397

File: rust/operator-binary/src/superset_controller.rs [hunk 1]
```diff
@@ -713,7 +714,7 @@ fn build_server_rolegroup_statefulset(
             .with_label("restarter.stackable.tech/enabled", "true")
             .build(),
         spec: Some(StatefulSetSpec {
-            pod_management_policy: Some("Parallel".to_string()),
+            pod_management_policy: Some("OrderedReady".to_string()),
```
Comment (maltesander) [inline rust/operator-binary/src/superset_controller.rs:717]: Whats the reason for this?

File: rust/operator-binary/src/superset_controller.rs [hunk 2]
```diff
@@ -628,19 +599,33 @@ fn build_server_rolegroup_statefulset(
...
         .image_from_product_image(resolved_product_image)
         .add_container_port("http", APP_PORT.into())
         .add_volume_mount(CONFIG_VOLUME_NAME, CONFIG_DIR)
         .add_volume_mount(LOG_CONFIG_VOLUME_NAME, LOG_CONFIG_DIR)
         .add_volume_mount(LOG_VOLUME_NAME, LOG_DIR)
+        .add_env_var_from_secret("ADMIN_USERNAME", &secret, "adminUser.username")
+        .add_env_var_from_secret("ADMIN_FIRSTNAME", &secret, "adminUser.firstname")
+        .add_env_var_from_secret("ADMIN_LASTNAME", &secret, "adminUser.lastname")
+        .add_env_var_from_secret("ADMIN_EMAIL", &secret, "adminUser.email")
+        .add_env_var_from_secret("ADMIN_PASSWORD", &secret, "adminUser.password")
```
Comment (maltesander) [inline rust/operator-binary/src/superset_controller.rs:614]:
```suggestion
    let secret = &superset.spec.cluster_config.credentials_secret;

    superset_cb
        .image_from_product_image(resolved_product_image)
        .add_container_port("http", APP_PORT.into())
        .add_volume_mount(CONFIG_VOLUME_NAME, CONFIG_DIR)
        .add_volume_mount(LOG_CONFIG_VOLUME_NAME, LOG_CONFIG_DIR)
        .add_volume_mount(LOG_VOLUME_NAME, LOG_DIR)
        .add_env_var_from_secret("ADMIN_USERNAME", secret, "adminUser.username")
        .add_env_var_from_secret("ADMIN_FIRSTNAME", secret, "adminUser.firstname")
        .add_env_var_from_secret("ADMIN_LASTNAME", secret, "adminUser.lastname")
        .add_env_var_from_secret("ADMIN_EMAIL", secret, "adminUser.email")
        .add_env_var_from_secret("ADMIN_PASSWORD", secret, "adminUser.password")
```

**Review by maltesander — CHANGES_REQUESTED** [review]
Nice work, just minor things. Changelog is missing.

File: rust/operator-binary/src/main.rs [hunk 3]
```diff
@@ -214,16 +140,17 @@ async fn main() -> anyhow::Result<()> {
...
                     watcher::Config::default(),
-                    move |superset_db| {
+                    move |superset_cluster| {
                         druid_connection_store_1
                             .state()
                             .into_iter()
                             .filter(move |druid_connection| {
-                                druid_connection.superset_name() == superset_db.name_unchecked()
+                                druid_connection.superset_name()
+                                    == superset_cluster.name_unchecked()
```
[the remark this answers is not in the document]
Comment (dervoeti) [reply rust/operator-binary/src/main.rs]: Thanks, fixed!

File: rust/operator-binary/src/superset_controller.rs (hunk 1 above)
Comment (maltesander) [reply rust/operator-binary/src/superset_controller.rs:717]: Could we add a comment for why it is set? Exactly the part with the DB init / upgrade?

Comment (dervoeti) [reply rust/operator-binary/src/superset_controller.rs:718]: Yes, good point, comment added.

**Review by maltesander — APPROVED** [review]
LGTM!

**Pull request merged**